cpp/src/arrow/util/tracing_internal.h

```diff
@@ -114,23 +117,38 @@ class SpanImpl {
 opentelemetry::trace::StartSpanOptions SpanOptionsWithParent(
     const util::tracing::Span& parent_span);
 
-#define START_SPAN(target_span, ...)                                                \
-  auto opentelemetry_scope##__LINE__ =                                              \
-      ::arrow::internal::tracing::GetTracer()->WithActiveSpan(                      \
-          target_span                                                               \
-              .Set(::arrow::util::tracing::Span::Impl{                              \
-                  ::arrow::internal::tracing::GetTracer()->StartSpan(__VA_ARGS__)}) \
-              .span)
+#define FIRST_ARG(first, ...) first
+
+#define START_COMPUTE_SPAN(target_span, ...) \
+  START_SPAN(target_span, __VA_ARGS__);      \
+  START_SPAN(target_span, FIRST_ARG(__VA_ARGS__), {GET_MEMORY_POOL_INFO});
+
+#define START_COMPUTE_SPAN_WITH_PARENT(target_span, parent_span, ...)      \
+  START_SPAN_WITH_PARENT(target_span, parent_span, __VA_ARGS__);           \
+  START_SPAN_WITH_PARENT(target_span, parent_span, FIRST_ARG(__VA_ARGS__), \
+                         {GET_MEMORY_POOL_INFO});
+
+#define START_SPAN(target_span, ...)                                                  \
+  {                                                                                   \
+    auto opentelemetry_scope##__LINE__ =                                              \
+        ::arrow::internal::tracing::GetTracer()->WithActiveSpan(                      \
+            target_span                                                               \
+                .Set(::arrow::util::tracing::Span::Impl{                              \
+                    ::arrow::internal::tracing::GetTracer()->StartSpan(__VA_ARGS__)}) \
+                .span);                                                               \
+  }
```

Comment:
I'm not sure how those work in OpenTelemetry, but this seems to close the scope at the end of the macro, is this right? @lidavidm

Reply:
Yes, we shouldn't create the OT scope in a C++ scope here, because the OT scope object controls how long a span is active.

Reply:
@mbrobbel i put the scopes because `auto opentelemetry_scope##__LINE__` when i use two START_SPAN it gives and conflict error in the variable name declaration, the opentelemetry_scope seems to be unused and when I put the scope for avoid the variable name declaration conflict seems to work fine https://paste.ofcode.org/gGUZ3Kk9H9HuyeLdJCpCX7

Reply:
The scope is an RAII guard, you wouldn't generally actually use it. But effectively it makes no sense to put a scope inside a C++ block by itself.